WordPress/gutenberg
# Allow using CSS level 4 viewport-relative units

## What?
Adds support for viewport-relative, CSS level-4 units.
These units are not exposed anywhere by default, but if a user enables them via `theme.json`, they should be available.

All these units have good browser support (see https://caniuse.com/viewport-unit-variants) and will allow designers to build more versatile interfaces.

Fixes #52441

## Why?

Because we should not explicitly forbid the use of CSS units - which is what happens if they are not defined in our lists of available CSS units.

## How?


## Testing Instructions
* On a site using the twentytwentythree theme (or any other block theme), edit the `theme.json` file. In `settings.spacing.units` we have an array of units that the theme supports. In that array, add some of the units defined in this PR (like for example `svmin`, `svmax`)
* Go to the editor, add a paragraph and change its padding. In the dropdown of units, you should now see the units you added (this is not the case in trunk)
* Save a value using the new units, and ensure that the editor preview changes accordingly, and the padding does indeed change
* Save the post and confirm that the frontend uses the new values as well

### Testing Instructions for Keyboard
Nothing specific for keyboard users. This change allows selecting more units, and does not change the existing selectors so nothing changes in the UI/UX.

## Screenshots or screencast 
<img width="372" alt="Screenshot 2023-09-14 at 10 10 15 AM" src="https://github.com/WordPress/gutenberg/assets/588688/ce983d26-df1f-41e4-b05c-de3b46bc6f9d">
